Source/WebCore/html/parser/HTMLParserScheduler.cpp

```diff
@@ -120,6 +120,9 @@ bool HTMLParserScheduler::shouldYieldBeforeExecutingScript(const ScriptElement*
     if (UNLIKELY(m_documentHasActiveParserYieldTokens))
         return true;
 
+    if (document->view() && !document->view()->hasEverPainted() && document->isLayoutPending())
```

Comment:
It's funny there is a comment above saying: 
> If we've never painted before and a layout is pending, yield prior to running
> scripts to give the page a chance to paint earlier.

Reply:
Yes :) I should probably move that comment down to the change.

Reply:
Thanks for reviewing!